### PR TITLE
New Chat & Economy API

### PR DESCRIPTION
**Pull request by Xxyuri2005xX**

### CODE REVIEW

File: bedwars-api/src/main/java/com/tomkeuper/bedwars/api/BedWars.java [hunk 1]
```diff
@@ -503,4 +505,13 @@ interface ScoreboardUtil {
     }
 
     IScoreboardService getScoreboardManager();
+
+    IEconomy getEconomy();
+
+    IChat getChat();
```
Comment (tomkeuper) [inline bedwars-api/src/main/java/com/tomkeuper/bedwars/api/BedWars.java]: getChatUtil()
getEconomyUtil()

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/support/vault/NoChat.java [hunk 2]
```diff
@@ -20,9 +20,10 @@
 
 package com.tomkeuper.bedwars.support.vault;
 
+import com.tomkeuper.bedwars.api.chat.IChat;
 import org.bukkit.entity.Player;
 
-public class NoChat implements Chat {
```
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/tomkeuper/bedwars/support/vault/NoChat.java:25]: I dont see that you removed the old Chat and Economy interfaces, is this correct?

**Review by tomkeuper — CHANGES_REQUESTED** [review]

File: bedwars-api/src/main/java/com/tomkeuper/bedwars/api/BedWars.java (hunk 1 above)
Comment (Xxyuri2005xX) [reply bedwars-api/src/main/java/com/tomkeuper/bedwars/api/BedWars.java]: ok, i can rename them

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/support/vault/NoChat.java (hunk 2 above)
Comment (Xxyuri2005xX) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/support/vault/NoChat.java:25]: I have not yet eliminated the old Chat and Economy classes but I can do it

Comment (tomkeuper) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/support/vault/NoChat.java:25]: They wont be used any more so clean up is good

**Comment by Xxyuri2005xX** [comment]
changes made

**Review by tomkeuper — APPROVED** [review]

**Comment by tomkeuper** [comment]
I will try merge in a sec

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/API.java [hunk 3]
```diff
@@ -447,4 +449,20 @@ public boolean isShuttingDown() {
...
+
+    @Override
+    public IChat getChatUtil() {
+        return BedWars.getChatSupport();
+    }
+
+    @Override
+    public void setEconomyAdapter(IEconomy economyAdapter) {
+        BedWars.setEconomy(economyAdapter);
+    }
```
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/tomkeuper/bedwars/API.java:466]: wouldnt it be better to move this into the economy util?

**Pull request merged**